src/main/java/com/github/manolo8/darkbot/gui/utils/inspector/ObjectInspectorUI.java

```diff
@@ -1,54 +1,106 @@
 package com.github.manolo8.darkbot.gui.utils.inspector;
 
+import com.github.manolo8.darkbot.core.BotInstaller;
+import com.github.manolo8.darkbot.core.manager.HeroManager;
 import com.github.manolo8.darkbot.core.utils.ByteUtils;
+import com.github.manolo8.darkbot.gui.MainGui;
 import com.github.manolo8.darkbot.utils.debug.ObjectInspector;
+import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
 import javax.swing.tree.DefaultTreeModel;
-import java.awt.*;
 import java.util.Objects;
 import java.util.function.BiConsumer;
+import java.util.function.Supplier;
 
 
 public class ObjectInspectorUI extends JFrame {
 
     public ObjectInspectorUI() {
         super("Object Inspector");
-        setLayout(new BorderLayout());
+        setLayout(new MigLayout());
         setSize(600, 600);
+        setIconImage(MainGui.ICON);
 
-        JTree treeView = new InspectorTree(new DefaultTreeModel(new ObjectTreeNode(
+        InspectorTree treeView = new InspectorTree(new DefaultTreeModel(new ObjectTreeNode(
                 new ObjectInspector.Slot("", "", null, 0, 0),
-                0, true)));
+                () -> 0L, true)));
 
-        JTextField addressField = new AddressField((name, addr) -> {
+        JComboBox<AddressEntry> addressCombo = new AddressCombo((name, addr) -> {
             DefaultTreeModel model = (DefaultTreeModel) treeView.getModel();
             model.setRoot(new ObjectTreeNode(
                     new ObjectInspector.Slot(name, "Object", null, 0, 8), addr, true));
         });
 
-        setLayout(new BorderLayout());
-        add(addressField, BorderLayout.NORTH);
-        add(new JScrollPane(treeView), BorderLayout.CENTER);
+        JSpinner delaySpinner = new JSpinner(new SpinnerNumberModel(250, 10, 1000, 10));
+        delaySpinner.addChangeListener(l -> treeView.setTimerDelay((Integer) delaySpinner.getValue()));
+
+        add(addressCombo, "grow");
+        add(delaySpinner, "wrap");
+        add(new JScrollPane(treeView), "push, grow, span");
     }
 
-    private static class AddressField extends JTextField {
+    private static class AddressCombo extends JComboBox<AddressEntry> {
 
-        public AddressField(BiConsumer<String, Long> onRootUpdate) {
+        public AddressCombo(BiConsumer<String, Supplier<Long>> onRootUpdate) {
             addActionListener(ae -> {
-                try {
-                    long object = Long.parseLong(getText(), 16);
+                Long object;
+                Object selectedItem = getSelectedItem();
+                if (selectedItem instanceof AddressEntry) {
+                    object = ((AddressEntry) selectedItem).address.get();
+                } else {
+                    object = tryParse((String) selectedItem, true);
+                    if (object == null) tryParse((String) selectedItem, false);
```

Comment:
this won't work, if a number looks decimal, eg: "1234" it will successfully parse as hex, but the value will be something else.
If anything you'd have to try to parse as decimal, and it'd fail under situations like "a123", then fall-back to hex.

I'd just make it so that 0x[whatever] is hex, check with startsWith or smthg, and based on that do one or the other